[CFX-4389] Enhance smoke tests handling for forked PRs and add status gate (#440)
datarobot-oss/cli: .github/workflows/README.md

- Auto-removes `run-smoke-tests` label after completion
- **Note:** Does NOT run installation tests (those run only on main/schedule to avoid testing unreleased code)

### `smoke-tests-gate.yaml`
Runs on all PRs targeting `main` to set the required **Smoke Tests** commit status:
- **Non-fork PRs**: auto-sets status to `success` (smoke tests are optional, run via labels/commands)
- **Fork PRs**: skips status creation (read-only token). The missing required check blocks merge until a maintainer runs (`/approve-smoke-tests`) or skips (`/skip-smoke-tests`) the tests

### `fork-smoke-tests.yaml`

> Triggered manually via `workflow_dispatch` by a maintainer (from the Actions tab):

Reviewer: ```suggestion
Triggered manually via `workflow_dispatch` by a maintainer, say with `/approve-smoke-tests`:
```
